packages/block-library/src/image/edit.native.js

```diff
@@ -363,6 +364,10 @@ export class ImageEdit extends Component {
 
 		setAttributes( { url: payload.mediaUrl, id: payload.mediaServerId } );
 		this.setState( { uploadStatus: UPLOAD_STATE_SUCCEEDED } );
+
+		this.setState( {
+			isAnimatedGif: payload.mediaUrl.toLowerCase().includes( '.gif' ),
```

Comment:
@illusaen, thanks for the updates so far! I think there's just one more flow I spotted that needs fixing before this seems good to go to me. 

`isAnimatedGif` is updated from within `finishMediaUploadWithSuccess`, but that function's only called when a new image is uploaded to a post, not when a previously uploaded image is selected from the Media Library. You'll see that the `.gif` I select from the Media Library doesn't get a badge for that reason:

https://user-images.githubusercontent.com/2998162/155749130-0a9f7dc9-af32-4227-abad-d0ecfec7278e.mov

It looks like a different approach for updating the state is warranted.

Reply:
@SiobhyB Added flow for images in WordPress Media Library!

https://user-images.githubusercontent.com/13263478/156117744-865b31ea-c673-4c75-953f-6fb64a0ac04c.mp4